### DISCUSSION TITLE
SideIntegralVariablePostprocessor over a Specified Boundary

### DISCUSSION THREAD

**Discussion by jsevic** [idaholab/moose#26424] [Q&A General] [answered]
Suppose I am using SideIntegralVariablePostprocessor over an edge of a 2-D object.

Is there a way to integrate only over a specified portion of this edge. For example, if the edge is y =10 for x = 0 to x = 50, could the integration be done only over x = 10 to x = 20?

Thank you!

**Comment by GiudGiud** [comment]
Hello

Two options I can think off:
- use a parsedGenerateSideset to create a subset of the edge in a sideset and then use that sideset in the postprocessor

- use the new mesh divisions system to create a grid excluding the portion you dont care about then make a **new** postprcoessor that integrates over a side in a mesh division

Guillaume

**Reply by jsevic** [reply]
OK thanks I figured I would try parsedGenerateSideset and began by modifying my current mesh statement to give it a name. That new code is shown too and I get this error, do I need to register something thx!


*** ERROR ***
Task add_mesh_generator is not registered to build MooseMesh derived objects


`[Mesh]
    type = GeneratedMesh
    dim = 2
    elem_type = QUAD4
    nx = 80
    ny = 20
    xmin = 0
    xmax = 40
    ymin = 0
    ymax = 10
    uniform_refine = 2
[]`

`[Mesh]
  [structure]
    type = GeneratedMesh
    dim = 2
    elem_type = QUAD4
    nx = 80
    ny = 20
    xmin = 0
    xmax = 40
    ymin = 0
    ymax = 10
    uniform_refine = 2
  []
[]`

**Reply by GiudGiud** [reply]
do this instead
```
[Mesh] 
  [gmg]
    type = GeneratedMeshGenerator
    dim = 2
    elem_type = QUAD4
    nx = 80
    ny = 20
    xmin = 0
    xmax = 40
    ymin = 0
    ymax = 10
  []
  uniform_refine = 2
[]
```

**Reply by GiudGiud** [reply]
the problem is that the old "all my paramenters directly in [Mesh]" syntax only works with a single mesh object. And here you want two, once for generating the mesh and one for modifying the sideset. So you need to update to the new nested MeshGenerator syntax
```
[Mesh]
  [name of the mesh generator]
    type ....
  []
  [name of the second mesh generator]
    type ....
  []
[]

**Reply by jsevic** [reply]
OK thanks I tried this below and I get an error about adding periodic BC's I am using these on two of my state variables?

*** ERROR ***
Couldn't auto-detect a paired boundary for use with periodic boundary conditions

`[Mesh] 
  [gmg]
    type = GeneratedMeshGenerator
    dim = 2
    elem_type = QUAD4
    nx = 80
    ny = 20
    xmin = 0
    xmax = 40
    ymin = 0
    ymax = 10
  []
  [sideset]
    type = ParsedGenerateSideset
    input = gmg
    combinatorial_geometry = 'y = 10 & 4 < x < 6'
    new_sideset_name = new_integration_domain
  []
  uniform_refine = 2
[]`

**Reply by GiudGiud** [reply]
this error is not coming from the mesh block. It s coming from your periodic BC block.
Can you pass more parameters to the Periodic BC block to avoid using the automatic dectection?

**Reply by jsevic** [reply]
OK I am looking at the documentation and want to make sure I understand setting PBC's. Are the primary and secondary names of edges in this particular example, 3 and 1 for the x-direction an these edges 40 units apart and 0 and 2 for the y-direction and these edges are also 40 units apart? Thx!

I also ran my code temporarily without PBS's and the `parsedGenerateSideset ` worked.


```
[BCs]
  [./Periodic]
    [./x]
      variable = u
      primary = 3
      secondary = 1
      translation = '40 0 0'
    [../]

    [./y]
      variable = u
      primary = 0
      secondary = 2
      translation = '0 40 0'
    [../]
  [../]
[]
```

**Reply by GiudGiud** [reply, marked as answer]
Primary and secondary are the names of the edges (sidesets) indeed. 
Can you use the names instead? Left and right should be easy to pair for example

**Reply by jsevic** [reply]
OK that works I and my results match the auto PBC however I noticed that if I set the new sideset to a small fraction of the initial sideset dimension the integrated current does not change, I was expecting that it would scale with the length of the integration domain. Any suggestions thx!

**Reply by GiudGiud** [reply]
so it's detecting more than the new sideset in a way? It's using the old sideset for the boundary condition?

Are you restricting it to a small fraction on both sides of the periodic boundary?

**Comment by jsevic** [comment]
I am integrating current density over the top edge of a 2-D thin film, the goal was to integrate current density only over a specified region of this top edge. This top edge is defined at y = 10 and over 0 < x <  10.

If the new sideset is define y = 10 and then 0 < x < 5 I was expecting to see current drop by approximately 2x but it remained the same as integrating over the old sideset.

How do restrict it to a small fraction on both sides of the periodic boundary? What does that mean? Thx!


```
  [sideset]
    type = ParsedGenerateSideset
    input = gmg
    combinatorial_geometry = 'y = 10 & 0 < x < 5'
    new_sideset_name = new_integration_domain
  []

[Postprocessors]
  [top_edge_current]
  type = SideIntegralVariablePostprocessor
  variable = J_mag
  boundary = new_integration_domain
  []
[]
```

**Reply by GiudGiud** [reply]
I am actually not sure fparser likes " 'y = 10 & 0 < x < 5'", for two reasons. Can you try this for the combinatorial geometry
```
 'y < 10.0000001 & y > 9.999999 & 0 < x & x < 5'
```

then can you check the mesh file to make sure it looks like half of the sideset

**Reply by GiudGiud** [reply]
btw I see your posts are formatted with a single backquote. If you use three backquotes instead it looks a lot better. I just edited your post

**Reply by jsevic** [reply]
OK cool thank you let me try and I will use 3x quotes I was wondering how to do that.

**Comment by jsevic** [comment]
OK that works like I expected thank you!
